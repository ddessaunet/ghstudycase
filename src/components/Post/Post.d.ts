import { User } from 'services/UserService.d';
import { ReactNode } from 'react';

export interface Props {
  image: string;
  likes: number;
  tags: string[];
  text: string;
  publishDate: string;
  action: (tag) => void;
  children?: ReactNode;
}
