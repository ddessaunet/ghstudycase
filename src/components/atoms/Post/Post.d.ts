import { User } from 'services/UserService';
import { ReactNode } from 'react';
import { AxiosRequestConfig } from 'axios';

export interface Props {
  image: string;
  likes: number;
  tags: string[];
  text: string;
  publishDate: string;
  action: (tag) => void;
  request?: () => AxiosRequestConfig;
  children?: ReactNode;
}
