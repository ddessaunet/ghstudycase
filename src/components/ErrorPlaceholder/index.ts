export type { Props as ErrorPlaceholderProps } from './ErrorPlaceholder.d';
export { ErrorPlaceholder } from './ErrorPlaceholder';
