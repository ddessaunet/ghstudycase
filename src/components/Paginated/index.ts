export type { Props as PaginatedProps } from './Paginated.d';
export { Paginated } from './Paginated';
