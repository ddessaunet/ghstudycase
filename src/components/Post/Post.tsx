import React from 'react';

const Post = () => <div />;

export default Post;
