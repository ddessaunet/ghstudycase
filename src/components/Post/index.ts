export type { Props as PostProps } from './Post.d';
export { default as Post } from './Post';
