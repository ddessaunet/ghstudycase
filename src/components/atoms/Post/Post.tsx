import { Box, Container, Flex, Img, Link, Text } from '@chakra-ui/react';
import React from 'react';
import { Props } from './Post.d';
import { ReactComponent as LikeIcon } from 'assets/like.svg';

export const Post = ({
  image,
  likes,
  tags,
  text,
  publishDate,
  action,
  children,
}: Props) => (
  <Container
    border="solid"
    borderWidth="5px"
    padding="10px"
    borderStyle="groove"
  >
    <Flex flexDirection="column">
      <Flex>
        <Box marginX="10px">
          <Img src={image} alt="" w="150px" />
        </Box>
        <Flex marginX="1px" w="100%">
          <Flex
            flexDirection="column"
            alignItems="flex-start"
            marginY="10px"
            marginBottom="auto"
            minW="50%"
          >
            <Box>
              <Text fontWeight="800">{publishDate}</Text>
            </Box>
            <Box>
              <Text fontWeight="800">{text}</Text>
            </Box>
            <Flex>
              {tags.map((tag: string) => (
                <Link
                  key={tag}
                  fontWeight="bold"
                  m="10px"
                  p="5px"
                  backgroundColor="blue.300"
                  textColor="white"
                  borderRadius="8px"
                  onClick={() => action(tag)}
                >
                  {tag}
                </Link>
              ))}
            </Flex>
          </Flex>
          <Box marginLeft="auto">
            <LikeIcon width="20px" />
            <Text display="inline" fontWeight="800">
              {likes}
            </Text>
          </Box>
        </Flex>
      </Flex>
      {children}
    </Flex>
  </Container>
);
