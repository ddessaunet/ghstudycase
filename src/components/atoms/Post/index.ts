export type { Props as PostProps } from './Post.d';
export { Post } from './Post';
