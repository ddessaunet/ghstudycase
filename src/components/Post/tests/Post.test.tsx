import React from 'react';
import { render } from '@testing-library/react';
import { Post } from 'components/Post';
import { PostProps } from 'components/Post';
import { ChakraProvider } from '@chakra-ui/react';

describe('<Post />', () => {
  const props = {
    image: 'https://img.dummyapi.io/photo-1589469526983-0887ebe10072.jpg',
    likes: 2,
    tags: ['dog', 'water', 'pool'],
    text: 'Cooling off in the fountain white and black short ...',
    publishDate: '2020-05-14T16:28:10.135Z',
    action: () => {},
  };
  const renderComponent = (props: PostProps) =>
    render(
      <ChakraProvider>
        <Post {...props} />
      </ChakraProvider>,
    );

  it('should render and match snapshot', () => {
    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });
});
