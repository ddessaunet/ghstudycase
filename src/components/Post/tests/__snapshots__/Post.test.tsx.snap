// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-container css-140un8w"
      >
        <div
          class="css-j7qwjs"
        >
          <div
            class="css-k008qs"
          >
            <div
              class="css-djdnrm"
            >
              <img
                alt=""
                class="chakra-image css-1nbic85"
                src="https://img.dummyapi.io/photo-1589469526983-0887ebe10072.jpg"
              />
            </div>
            <div
              class="css-1dglqzs"
            >
              <div
                class="css-1o7kqr2"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    2020-05-14T16:28:10.135Z
                  </p>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-ty7r4z"
                  >
                    Cooling off in the fountain white and black short ...
                  </p>
                </div>
                <div
                  class="css-k008qs"
                >
                  <a
                    class="chakra-link css-16aod2b"
                  >
                    dog
                  </a>
                  <a
                    class="chakra-link css-16aod2b"
                  >
                    water
                  </a>
                  <a
                    class="chakra-link css-16aod2b"
                  >
                    pool
                  </a>
                </div>
              </div>
              <div
                class="css-zdpt2t"
              >
                <svg
                  width="20px"
                >
                  like.svg
                </svg>
                <p
                  class="chakra-text css-1usi692"
                >
                  2
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-container css-140un8w"
    >
      <div
        class="css-j7qwjs"
      >
        <div
          class="css-k008qs"
        >
          <div
            class="css-djdnrm"
          >
            <img
              alt=""
              class="chakra-image css-1nbic85"
              src="https://img.dummyapi.io/photo-1589469526983-0887ebe10072.jpg"
            />
          </div>
          <div
            class="css-1dglqzs"
          >
            <div
              class="css-1o7kqr2"
            >
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ty7r4z"
                >
                  2020-05-14T16:28:10.135Z
                </p>
              </div>
              <div
                class="css-0"
              >
                <p
                  class="chakra-text css-ty7r4z"
                >
                  Cooling off in the fountain white and black short ...
                </p>
              </div>
              <div
                class="css-k008qs"
              >
                <a
                  class="chakra-link css-16aod2b"
                >
                  dog
                </a>
                <a
                  class="chakra-link css-16aod2b"
                >
                  water
                </a>
                <a
                  class="chakra-link css-16aod2b"
                >
                  pool
                </a>
              </div>
            </div>
            <div
              class="css-zdpt2t"
            >
              <svg
                width="20px"
              >
                like.svg
              </svg>
              <p
                class="chakra-text css-1usi692"
              >
                2
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
